packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1228,24 +1229,34 @@ function handleError(root, thrownValue): void {
         return;
       }
 
-      if (enableProfilerTimer && workInProgress.mode & ProfileMode) {
+      if (enableProfilerTimer && erroredWork.mode & ProfileMode) {
         // Record the time spent rendering before an error was thrown. This
         // avoids inaccurate Profiler durations in the case of a
         // suspended render.
-        stopProfilerTimerIfRunningAndRecordDelta(workInProgress, true);
+        stopProfilerTimerIfRunningAndRecordDelta(erroredWork, true);
       }
 
       throwException(
         root,
-        workInProgress.return,
-        workInProgress,
+        erroredWork.return,
+        erroredWork,
         thrownValue,
         renderExpirationTime,
       );
-      completeUnitOfWork(workInProgress);
+      completeUnitOfWork(erroredWork);
     } catch (yetAnotherThrownValue) {
       // Something in the return path also threw.
       thrownValue = yetAnotherThrownValue;
+      erroredWork = workInProgress;
+      if (
+        erroredWork !== null &&
+        (erroredWork.effectTag & Incomplete) !== NoEffect
```

Comment:
I don't think it matters, but I expected this to be `DidCapture`, since that's the one that says we attempted a second pass. Whereas `Incomplete` is set on all the fibers in between the source and the boundary.

Would it work to compare `workInProgress` to `erroredWork`?  And if they are the same, we know something in the error handling path went wrong?

Reply:
Sorry, I meant stash `workInProgress` before re-entering `completeUnitOfWork` and then compare to that.

Reply:
It's not just boundaries that can cause this so DidCapture doesn't work but comparing workInProgress to erroredWork should work.